src/app/bungie-api/http-client.ts

```diff
@@ -158,7 +158,10 @@ export function createHttpClient(
       credentials: withCredentials ? 'include' : 'omit',
     });
     const response = await fetchFunction(fetchOptions);
-    const data: ServerResponse<unknown> = await response.json();
+    let data: ServerResponse<unknown> | undefined;
+    try {
+      data = await response.json();
+    } catch {}
```

Comment:
why is completely silent failure, preferable to throwing an error indicating that the response was invalid/not JSON?
i **really strongly disagree** with the major shift of allowing `HttpClient` to return undefined.
none of the bungie-api-ts functions account for this at all, they presume `HttpClient`'s `any` is their data type, assuming no error was thrown.

Reply:
We can throw something if data doesn't end up returning anything - the core problem is the error thrown by parsing JSON causes this to skip checking for HTTP status codes.